File: Prompt Engineering/Lesson6_Prompt_to_Project_Showcase/Activity1_Choose_Your_Own_Game.js
// ========================================================================
// Prompt Engineering - Lesson 6: Prompt-to-Project Showcase (Capstone)
// Activity 1: Choose Your Own Game — Project 1: Meteor Dodger
// Platform: Code.org Game Lab (JavaScript)
// ========================================================================

// ------------------------------------------------------------------------
// Game Description:
// Pilot your spaceship across deep space! Dodge incoming meteors from above
// while collecting glowing energy stars. Survive for 30 seconds and score
// at least 8 stars to achieve mission victory!
// ------------------------------------------------------------------------

// Background sprite
var bg = createSprite(200, 200);
bg.setAnimation("background_space_1");

// Player spaceship sprite
var ship = createSprite(200, 340);
ship.setAnimation("retro_ship_1");
ship.scale = 0.35;

// Tumbling meteor obstacle
var meteor = createSprite(randomNumber(40, 360), -30);
meteor.setAnimation("meteor_1");
meteor.scale = 0.4;
var meteorSpeed = 6;

// Collectible star
var star = createSprite(randomNumber(40, 360), -20);
star.setAnimation("star_1");
star.scale = 0.25;
var starSpeed = 4;

// Game state variables
var score = 0;
var lives = 3;
var timer = 30;

function draw() {
  background("black");

  // Active gameplay while timer > 0 and lives > 0
  if (timer > 0 && lives > 0) {
    // 1. Player controls (Left / Right with screen boundary constraint)
    if (keyDown("left")) {
      ship.x -= 6;
    }
    if (keyDown("right")) {
      ship.x += 6;
    }
    ship.x = constrain(ship.x, 30, 370);

    // 2. Obstacle movement: Meteor falls
    meteor.y += meteorSpeed;
    if (meteor.y > 430) {
      meteor.y = -30;
      meteor.x = randomNumber(40, 360);
    }

    // 3. Collectible movement: Star falls
    star.y += starSpeed;
    if (star.y > 420) {
      star.y = -20;
      star.x = randomNumber(40, 360);
    }

    // 4. Collision check: Ship catches Star (+1 Score)
    if (ship.isTouching(star)) {
      score++;
      star.y = -20;
      star.x = randomNumber(40, 360);
    }

    // 5. Collision check: Ship hits Meteor (-1 Life)
    if (ship.isTouching(meteor)) {
      lives--;
      meteor.y = -30;
      meteor.x = randomNumber(40, 360);
    }

    // 6. Countdown Timer (30 FPS)
    if (World.frameCount % 30 === 0) {
      timer--;
    }

    // Draw all active sprites
    drawSprites();

    // 7. Polished HUD Overlay
    fill("white");
    rect(0, 0, 400, 40);

    fill("black");
    textSize(18);
    text("Stars: " + score, 20, 25);
    text("Lives: " + lives, 160, 25);
    text("Time: " + timer + "s", 300, 25);
  }

  // End of Game Screens
  else {
    drawSprites();

    fill("yellow");
    textSize(30);

    var isWin = (timer <= 0 && lives > 0 && score >= 8);

    if (isWin) {
      text("Mission Success! 🚀", 70, 200);
      textSize(20);
      fill("white");
      text("You collected " + score + " stars and survived!", 50, 240);
    } else {
      text("Game Over!", 120, 200);
      textSize(20);
      fill("white");
      if (lives <= 0) {
        text("Your ship took too much damage!", 65, 240);
      } else {
        text("Time ran out! Final Score: " + score + " (Needed 8)", 45, 240);
      }
    }
  }
}

File: Prompt Engineering/Lesson6_Prompt_to_Project_Showcase/Activity1_Choose_Your_Own_Game.py
"""
Prompt Engineering - Lesson 6: Prompt-to-Project Showcase
Activity 1: Choose Your Own Game (Capstone Launcher)

Interactive Python Launcher for the Capstone Game Showcase.
Allows students to play simulated games, review prompt breakdowns, and choose themes.
"""

import random
import time

def banner(title):
    print("\n" + "=" * 70)
    print(f"   🚀 {title.upper()} 🎮")
    print("=" * 70)

def play_meteor_dodger():
    banner("Meteor Dodger Terminal Game")
    print("Mission: Dodge Meteors (☄️) and catch Stars (⭐)!")
    print("Controls: 'a' (left), 'd' (right), Enter (stay)\n")

    ship_x = 3
    stars = 0
    lives = 3
    rounds = 12
    width = 7

    while rounds > 0 and lives > 0 and stars < 5:
        meteor_x = random.randint(0, width - 1)
        star_x = random.randint(0, width - 1)
        while star_x == meteor_x:
            star_x = random.randint(0, width - 1)

        print("\n" + "—" * 35)
        print(f"🛰️ STARS: {stars}/5 | LIVES: {'❤️' * lives} | TIME: {rounds}s")
        print("—" * 35)

        sky = ["·"] * width
        sky[meteor_x] = "☄️"
        sky[star_x] = "⭐"
        print(f"Space:  {' '.join(sky)}")

        ground = ["·"] * width
        ground[ship_x] = "🚀"
        print(f"Ship:   {' '.join(ground)}")

        move = input("\nControl (a/d): ").strip().lower()
        if move == 'a' and ship_x > 0:
            ship_x -= 1
        elif move == 'd' and ship_x < width - 1:
            ship_x += 1

        if ship_x == meteor_x:
            lives -= 1
            print("💥 HIT! Meteor struck your shields! (-1 Life)")
        elif ship_x == star_x:
            stars += 1
            print("✨ COLLECTED! Star absorbed! (+1 Star)")
        else:
            print("🌌 Maneuver successful, drifting safely.")

        rounds -= 1
        time.sleep(0.3)

    print("\n" + "=" * 40)
    if stars >= 5 and lives > 0:
        print("🏆 MISSION SUCCESS! 🚀")
        print(f"Congratulations! Stars: {stars}, Remaining Lives: {lives}")
    else:
        print("💀 MISSION FAILED!")
        if lives <= 0:
            print("Your ship was destroyed by meteors!")
        else:
            print(f"Time ran out! Final Stars: {stars}/5")
    print("=" * 40)

def view_all_options():
    banner("The 7 Capstone Menu Themes")
    themes = [
        ("1. Meteor Dodger 🚀", "Spaceship avoids falling meteors and collects energy stars."),
        ("2. Jungle Escape 🐒", "Monkey jumps over rocks and logs to collect bananas."),
        ("3. Animal Rescue 🧺", "Basket catches falling pets with scoring and time limits."),
        ("4. Mystery Door Challenge 🚪", "Click the correct key to unlock the magical castle portal."),
        ("5. Race to the Finish 🏎️", "Racecar steers through city traffic to beat the timer."),
        ("6. Underwater Explorer 🤿", "Diver collects shiny pearls while avoiding sharks."),
        ("7. Sky Jumper ☁️", "Hero leaps across floating cloud platforms without falling.")
    ]
    for name, desc in themes:
        print(f"\n👉 {name}")
        print(f"   {desc}")

def main():
    while True:
        banner("Prompt-to-Project Showcase Studio")
        print("1. Play Meteor Dodger Terminal Arcade")
        print("2. Explore All 7 Capstone Project Themes")
        print("3. Exit")

        c = input("\nEnter choice (1-3): ").strip()
        if c == "1":
            play_meteor_dodger()
        elif c == "2":
            view_all_options()
        elif c == "3":
            print("\nCongratulations on completing the Prompt Engineering Course! 🎓\n")
            break
        else:
            print("Invalid selection.")
        input("\nPress Enter to return to menu...")

if __name__ == "__main__":
    main()
